generate_lora.py

```diff
@@ -0,0 +1,149 @@
+import sys
+import time
+import warnings
+from pathlib import Path
+from typing import Optional
+
+import lightning as L
+import torch
+
+from generate import generate
+from lit_llama import Tokenizer
+from lit_llama.adapter import LLaMA, LLaMAConfig
+from lit_llama.lora import lora, lora_state_dict
+from lit_llama.utils import EmptyInitOnDevice
+from scripts.prepare_alpaca import generate_prompt
+
+lora_r = 8
+lora_alpha = 16
+lora_dropout = 0.05
+
+
+def main(
+    prompt: str = "What food do lamas eat?",
+    input: str = "",
+    lora_path: Optional[Path] = None,
+    pretrained_path: Optional[Path] = None,
+    tokenizer_path: Optional[Path] = None,
+    quantize: Optional[str] = None,
+    dtype: str = "float32",
+    max_new_tokens: int = 100,
+    top_k: int = 200,
+    temperature: float = 0.8,
+    accelerator: str = "auto",
+) -> None:
+    """Generates a response based on a given instruction and an optional input.
+    This script will only work with checkpoints from the instruction-tuned LoRA model.
+    See `finetune_lora.py`.
+
+    Args:
+        prompt: The prompt/instruction (Alpaca style).
+        lota_path: Path to the checkpoint with trained LoRA weights, which are the output of
+            `finetune_lora.py`.
+        input: Optional input (Alpaca style).
+        pretrained_path: The path to the checkpoint with pretrained LLaMA weights.
+        tokenizer_path: The tokenizer path to load.
+        quantize: Whether to quantize the model and using which method:
+            ``"llm.int8"``: LLM.int8() mode,
+            ``"gptq.int4"``: GPTQ 4-bit mode.
+        dtype: The dtype to use during generation.
+        max_new_tokens: The number of generation steps to take.
+        top_k: The number of top most probable tokens to consider in the sampling process.
+        temperature: A value controlling the randomness of the sampling process. Higher values result in more random
+            samples.
+        accelerator: The hardware to run on. Possible choices are:
+            ``"cpu"``, ``"cuda"``, ``"mps"``, ``"gpu"``, ``"tpu"``, ``"auto"``.
+    """
+    if not lora_path:
+        lora_path = Path("out/adapter/alpaca/lit-llama-lora-finetuned.pth")
```

Comment:
```suggestion
        lora_path = Path("out/lora/alpaca/lit-llama-lora-finetuned.pth")
```
And the finetune_lora is a bit different from finetune-alpaca as we don't write an end-of-trianing checkpoint

Reply:
I did add saving in finetune_alpaca.py